src/main/java/us/spaceclouds42/zones/mixin/FallingBlockEntityMixin.java

```diff
@@ -0,0 +1,54 @@
+package us.spaceclouds42.zones.mixin;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.FallingBlockEntity;
+import net.minecraft.world.GameRules;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import us.spaceclouds42.zones.data.ZoneManager;
+import us.spaceclouds42.zones.data.spec.PosD;
+import us.spaceclouds42.zones.data.spec.Zone;
+
+@Mixin(FallingBlockEntity.class)
+public abstract class FallingBlockEntityMixin extends Entity {
```

Comment:
Why do you extend entity?

Reply:
Because I access fields/methods of `Entity` in this mixin, and you can only shadow members of the class you are mixing in to.